src/workerd/api/sql.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include <workerd/jsg/string.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
```

Comment:
So compared to the [sqlite type system](https://www.sqlite.org/datatype3.html), I guess we're folding INTEGER into REAL here, which I suppose we have to because JavaScript, but that seems fine.

What about NULL, though?

I guess to support NULL we need to wrap the whole thing in a Maybe, i.e. `kj::Maybe<kj::OneOf<...>>`.

Reply:
One problem with cursors is we will need to have protection against trying to execute the same prepared statement twice concurrently. Maybe we should block concurrent queries in general, or maybe we should actually automatically make a second copy of the prepared statement under the hood...

Reply:
Yeah I ran into issues trying to add nullptr_t there. The problem is that I would have to implement `kj::Maybe<nullptr_t> tryUnwrap(...)` in JSG and `kj::Maybe<nullptr_t>` has an ambiguous constructor between `Maybe(decltype(nullptr)): ptr(nullptr) {}` and `Maybe(const T& t): ptr(t) {}`. I ripped out the nullptr constructor and fixed it up everywhere, but then even with that when I would return nullptr, JSG would convert it into a string of "null". I didn't attempt to trace it much further than that, but can certainly look into it.

Reply:
Hmm I just realized you meant wrap the entire thing in a Maybe, interesting idea!

Reply:
> One problem with cursors is we will need to have protection against trying to execute the same prepared statement twice concurrently. Maybe we should block concurrent queries in general, or maybe we should actually automatically make a second copy of the prepared statement under the hood...

I think maybe both? We absolutely should limit the number of concurrent queries that can be run, but also I think it ought to still be possible to run two different instances of the same prepared statement within that concurrency limit.